pageserver/src/tenant.rs

```diff
@@ -525,54 +505,38 @@ impl Tenant {
         .context("merge_local_remote_metadata")?
         .to_owned();
 
-        let timeline = {
+        let timeline = self.create_timeline_struct(
+            timeline_id,
+            up_to_date_metadata,
+            ancestor.clone(),
+            remote_client,
+            init_order,
+        )?;
+        let new_disk_consistent_lsn = timeline.get_disk_consistent_lsn();
+        anyhow::ensure!(
+            new_disk_consistent_lsn.is_valid(),
+            "Timeline {tenant_id}/{timeline_id} has invalid disk_consistent_lsn"
+        );
+        timeline
+            .load_layer_map(new_disk_consistent_lsn)
```

Comment:
you call the timeline object with a parameter that is already part of it (get_disk_consistent_lsn()) I didn't see any other call site for it 
so you can just have timeline::load_layer_map() that will use its internal state for the disk consistent lsn

this is also a good place to have timeline::ensure_disk_consistent_lsn_valid() method instead of  lines 515 - 519

Reply:
Good catch, yeah, `load_layer_map` is free-standing now.

I think I'll do the changes as a follow-up to not further clutter the diff. Await a PR :)